test/e2e/storage/non_csi_volumes.go

```diff
@@ -0,0 +1,1493 @@
+		if err == nil {
+			framework.Logf("Cinder volume %s deleted", name)
+			return nil
+		} else {
+			framework.Logf("Failed to delete volume %s: %v", name, err)
+		}
+	}
+	framework.Logf("Giving up deleting volume %s: %v\n%s", name, err, string(output[:]))
+	return err
+}
+
+// GCE
+type gceDriver struct {
+	volumeName string
+
+	driverInfo driverInfo
+}
+
+func initGceDriver(f *framework.Framework, config framework.VolumeTestConfig, node v1.Node) testDriver {
+	framework.SkipUnlessProviderIs("gce", "gke")
+
+	ns := f.Namespace
+	fsGroup := int64(1234)
+
+	// PD will be created in framework.TestContext.CloudConfig.Zone zone,
+	// so pods should be also scheduled there.
+	config.NodeSelector = map[string]string{
+		kubeletapis.LabelZoneFailureDomain: framework.TestContext.CloudConfig.Zone,
+	}
+
+	return &gceDriver{
+		driverInfo: driverInfo{
+			name:    "gce",
+			fsGroup: &fsGroup,
+			podSec: v1.PodSecurityContext{
+				FSGroup: &fsGroup,
+			},
+			privilegedSecurityContext: true,
+			maxFileSize:               fileSizeMedium,
+			supportedFsType: map[string]bool{
+				"":     true, // Default fsType
+				"ext2": true,
+				"ext3": true,
+				"ext4": true,
+				"xfs":  true,
+			},
+			isBlockSupported: false,
+			expectedContent:  "Hello from GCE from namespace " + ns.Name,
+			needInjectHTML:   true,
+			testFile:         "gce_io_test",
+			f:                f,
+			config:           config,
+			node:             node,
+		},
+	}
+}
+
+func (g *gceDriver) getDriverInfo() driverInfo {
+	return g.driverInfo
+}
+
+func (g *gceDriver) getVolumeSource(readOnly bool, fsType string) *v1.VolumeSource {
+	skipIfFsTypeNotSupported(g.driverInfo, fsType)
+	volSource := v1.VolumeSource{
+		GCEPersistentDisk: &v1.GCEPersistentDiskVolumeSource{
+			PDName:   g.volumeName,
+			ReadOnly: readOnly,
+		},
+	}
+	if fsType != "" {
+		volSource.GCEPersistentDisk.FSType = fsType
+	}
+	return &volSource
+}
+
+func (g *gceDriver) getPersistentVolumeSource(readOnly bool, fsType string) *v1.PersistentVolumeSource {
+	// TODO: implement this method propery
+	skipIfFsTypeNotSupported(g.driverInfo, fsType)
+	return nil
+}
+
+func (g *gceDriver) getDynamicProvisionStorageClass(fsType string) *storagev1.StorageClass {
+	skipIfFsTypeNotSupported(g.driverInfo, fsType)
+	provisioner := "kubernetes.io/gce-pd"
+	parameters := map[string]string{}
+	if fsType != "" {
+		parameters["fsType"] = fsType
+	}
+	ns := g.driverInfo.f.Namespace.Name
+	suffix := fmt.Sprintf("%s-sc", g.driverInfo.name)
+
+	return getStorageClass(provisioner, parameters, ns, suffix)
+}
+
+func (g *gceDriver) createDriver() {
+	By("creating a test gce pd volume")
+	var err error
+	g.volumeName, err = framework.CreatePDWithRetry()
+	Expect(err).NotTo(HaveOccurred())
+}
+
+func (g *gceDriver) cleanupDriver() {
+	config := g.driverInfo.config
+	// The volume would be mounted on config.clientNodeName
+	detachAndDeletePDs(g.volumeName, []types.NodeName{types.NodeName(config.ClientNodeName)})
```

Comment:
I don't think the cleanup should be doing detach.

Usually in a test, when the pod is terminated, then the volume automatically gets detached by Kubernetes.

Reply:
I'm not sure why but it seems to work well, now. 
I will ask you for your help again, once the issue is reproduced.